include/vectorSIMD.h

```diff
@@ -324,6 +324,12 @@ namespace core
 
 		inline vectorSIMD_32<T> operator-(T val) const { return (*this)-vectorSIMD_32<T>(val); }
 		inline vectorSIMD_32<T>& operator-=(T val) { return ( (*this) -= vectorSIMD_32<T>(val) ); }
+
+		inline vectorSIMD_32<T>& operator>>(uint32_t val) { _mm_store_si128(reinterpret_cast<__m128i*>(pointer), _mm_srai_epi32(Base::getAsRegister(),val)); return (*this); }
+
+		//comparison
+		inline vector4db_SIMD operator<(const vectorSIMDf& other) const { return _mm_cmplt_epi32(getAsRegister(), other.getAsRegister()); }

```

Comment:
why are you doing a compare against a vectorSIMDf?

Reply:
also this comparison op only works for signed ints

You'd need to use CRTP and make vectorSIMD_32<T> a common base class (as `vectorSIMD_32<VECTOR_TYPE<T> >`) for actual `vectorSIMDu32` and `vectorSIMDi32` (instead of them being typedefs as they are now)

Reply:
> why are you doing a compare against a vectorSIMDf?

oops